packages/paste-core/components/chat-log/src/ChatLog.tsx

```diff
@@ -2,17 +2,60 @@ import * as React from 'react';
 import * as PropTypes from 'prop-types';
 import {Box, safelySpreadBoxProps} from '@twilio-paste/box';
 import type {BoxElementProps} from '@twilio-paste/box';
+import {useTransition, animated, useReducedMotion} from '@twilio-paste/animation-library';
+import {uid} from '@twilio-paste/uid-library';
+
+const AnimatedChat = animated(Box);
+type StyleProps = React.ComponentProps<typeof AnimatedChat>['style'];
 
 export interface ChatLogProps {
   children?: React.ReactNode;
   element?: BoxElementProps['element'];
 }
 
+const isReactElement = (item: React.ReactNode): item is React.ReactElement => {
+  return (item as React.ReactElement)?.props !== undefined;
+};
+
+const isInboundChat = (item: React.ReactNode): boolean => {
+  return isReactElement(item) && item?.props?.variant === 'inbound';
```

Comment:
Yeah, this might start getting to why the toast hook is data driven and not jsx driven, because you don't need to necessarily inspect what is passed to you to figure out what to do. An example of that is toast variant, it's explicitly set on the object, so you don't need to look at passed in children, which you don't own technically, to figure out any specific knowledge